### PR TITLE
Feature/search transaction by user and type

### PR DESCRIPTION
**Pull request by PuchalskiMateusz**
PLEAS MERGE AFTER PREVIOUS PULL REQUESTS
aDDED TO METHOD TO SEARCH TRANSACTION FOR SPECIFIC USER, CHOOSEN BY EMAIL, BY TYPE

### CODE REVIEW

**Comment by MichalMichanski** [comment]
Pulled and tested, hate to say but i don't think i have anything id like to add, looks good

File: TheBTeam.ConsoleApp/ConsoleFactory.cs [hunk 1]
```diff
@@ -125,6 +125,25 @@ private static string GetStringInput(string name, int minLength)
...
+                }
+
+                Console.Write($"{name}: ");
+                input = Console.ReadLine()?.Trim();
+                if (input == null || input.Length < minLength || input.Any(char.IsDigit) || input.Any(char.IsWhiteSpace))
+                    Console.WriteLine($"Invalid data. {name} should have at least {minLength} char long and in correct format Retry!");
+                else
+                    return input;
+            }
+        }
```
Comment (MichalMichanski) [inline TheBTeam.ConsoleApp/ConsoleFactory.cs:146]: good stuff

File: TheBTeam.ConsoleApp/MainMenu.cs [hunk 2]
```diff
@@ -100,20 +101,26 @@ public static void ShowMainMenu()
...
+                    TransactionViewer.UserAndTypeTransaction(tmpListUsers.UsersList, tmpListTransactions.TransactionsList);
+                    Console.ReadKey();
+                }
                 else if (mainMenuItem[currentItem] == ("Edit existing user"))//there used to be like 3 paragraphs here regarding my reasoning, but fuck that, basically edit user wont work once we pass a tmplist
-                    //to it directly its messy due to the way references are handled
-                    //or maybe more the way i think they are handled - lists being copied, and objects referred
-                    //bottom line is it works but there definitely is to be a nicer way to do this, ill ask patryk during refinements
-                    //Actually nvm im dumb, since below works i jsut understood sb wrong, ill get below to look decent when im done with homework
+                                                                             //to it directly its messy due to the way references are handled
+                                                                             //or maybe more the way i think they are handled - lists being copied, and objects referred
```
Comment (MichalMichanski) [inline TheBTeam.ConsoleApp/MainMenu.cs:118]: Feel free to yeet the comment, it keeps getting added back whenever its present in sbs branch i think, logic applied turned out to actually make sense, althouh in an entirely different way, its obsolete

File: TheBTeam.ConsoleApp/TransactionViewer.cs [hunk 3]
```diff
@@ -89,9 +90,24 @@ public static int ViewTransactionEdit(List<Transaction> transactions)
...
+                return;
+               
+            var type = ConsoleFactory.GetTypeOfTransaction();
+
+            var transactionByUser = TransactionService.SearchTransactionByUser(user, transactions);
+            var transactionByType = TransactionService.SearchTransactionByType(type, transactionByUser);
+
+            ViewTransaction(transactionByType);
+        }
+
```
Comment (MichalMichanski) [inline TheBTeam.ConsoleApp/TransactionViewer.cs:111]: nice and clean

**Review by MichalMichanski — APPROVED** [review]

**Review by MaciejCVsGit — APPROVED** [review]

**Pull request merged**